rasterio/rio/windows.py

```diff
@@ -0,0 +1,171 @@
+
+        Yields
+        ------
+        dict
+            GeoJSON polygon feature.
+        """
+
+        self._src = src
+        self._bidx = bidx
+        self._precision = precision
+        self._geographic = geographic
+
+    @property
+    def bbox(self):
+        bounds = self._src.bounds
+        if self._geographic:
+            bounds = transform_bounds(self._src.crs, 'EPSG:4326', *bounds)
+        if self._precision > 0:
+            bounds = (round(v, self._precision) for v in bounds)
+        return tuple(bounds)
+
+    def __call__(self):
+        gen = self._src.block_windows(bidx=self._bidx)
+        for idx, (block, window) in enumerate(gen):
+            bounds = self._src.window_bounds(window)
+            if self._geographic:
+                bounds = transform_bounds(self._src.crs, 'EPSG:4326', *bounds)
+            if self._precision > 0:
+                bounds = (round(v, self._precision) for v in bounds)
+            xmin, ymin, xmax, ymax = bounds
+            yield {
+                'type': 'Feature',
+                'id': '{0}:{1}'.format(os.path.basename(self._src.name), idx),
+                'properties': {
+                    'block': json.dumps(block),
+                    'window': json.dumps(window)
+                },
+                'geometry': {
+                    'type': 'Polygon',
+                    'coordinates': [[
+                        (xmin, ymin),
+                        (xmin, ymax),
+                        (xmax, ymax),
+                        (xmax, ymin)
+                    ]]
+                }
+            }
+
+
+@click.command()
+@options.file_in_arg
+@options.output_opt
+@cligj.precision_opt
+@cligj.indent_opt
+@cligj.compact_opt
+@cligj.projection_geographic_opt
+@cligj.projection_projected_opt
+@cligj.sequence_opt
+@cligj.use_rs_opt
+@click.option(
+    '--bidx', type=click.INT, default=0,
+    help="Index of the band that is the source of shapes.")
+@click.pass_context
+def windows(
+        ctx, input, output, precision, indent, compact, projection, sequence,
+        use_rs, bidx):
+
+    """Write dataset windows as GeoJSON.
+
+    This command writes features describing a rasters internal tiling scheme,
+    which are often used to visually get a feel for how a windowed operation
+    might work.  Rasters that are not tiled internally can still have internal
+    blocks that are not square, and this command does not differentiate between
+    the two.
+
+    Output features have two JSON encoded properties: block and window.  Block
+    is a two element array like `[0, 0]` describing the window's position
+    in the input band's window layout.  Window is a two element array
+    containing two more two element arrays like `[[0, 256], [0, 256]]` and
+    describes the range of pixels the window covers in the input band.  Values
+    are JSON encoded for better interoperability.
+
+    The first band is used by default, but any other band can be specified
```

Comment:
@geowurster use of `bidx` is potentially confusing here; `bidx=1` is the first band, `bidx=0` indicates use the windows for the dataset in general (incidentally derived from the first band but all must match or an exception is thrown).  So maybe mentioning `first band is used by default` is unnecessary here, maybe instead 

```Block windows are extracted from the dataset (all bands must have matching block windows) by default, or from the band specified using the `--bidx` option:```